proposals/0052-experimental-dxil-ops.md

```diff
@@ -57,10 +57,97 @@ solution. Thus this proposal explicitly avoids addressing these issues:
+lacks process complexity it also lacks flexibility or resolution.
+
+The solution proposed is to implement the Top 16 Bit with the following restriction:
+ * Feature IDs must be either `0x0000` or `0x8000` until an undetermined time in
+   the future where the restriction will either be lifted or permentantly codified.
+
+Feature ID `0x0000` is to be used for stable opcodes that are published as part
+of the retail compiler. Feature ID `0x8000` is to be used for all expirmental
+opcodes. Feature ID `0x0000` must be used for stable opcodes to ensure that
+existing opcodes are not renumbered however the choice of ID `0x80000` may seem
+arbitrary. `0x8000` is selected for one key feature. The set bit is the top
+most bit fo the opcode space. Therefore all opcodes defined with either ID will
+match the underlying values they would have in the Top 1 bit proposal.
+
+With this restriction, the compiled DXIL is proposal agnostic. Either original
+proposal could have reasonably generated it. This is an imperfect compromise
+that has two key features:
+ * it unblocks high priority dependencies of the feature
+ * it cleanly collapses into either original proposal
+
+It is the intent of this proposal to serve as an expirement in its own right.
+At some point in the future this proposal should collapse into one of the above
+proposal in a non-breaking manner. After iterating through a number of
+development cycles, the level of complexity required to serve this feature will
+reveal itself. If the development challenges presented at the top of this
+document are resolved with the restrictions in place then the proposal will
+collapse into the 1bit solution and no further work is needed. However if the
+challenges persist then clearly the restrictions are too constraining. If this
+occurs then a new proposal shall be made to lift the restrictions allowing for
+any feature ID and addressing any required process changes which will collapse
+into the 16bit proposal.
+
+### Implementation Considerations
+
+#### DXC
+DXC has considerable constraints to align implementation with existing
+infrastructure. All of the bit width proposals are reasonable to implement
+but the other proposals are beyond reasonable scope. For the bit width proposals
+the infrastructure needs to be updated to support multiple op code tables. The
+detailed explination of that infrastructure is listed below.
+
+#### Clang
+Clang implementation for bit width proposals are trivial. Clang has a late
+lowering for convering high level notions of the DXIL ops into the specific
+opcode values. That mapping is arbitrary and can be implemented by simply
+setting the correct value for the opcode in `llvm/lib/Target/DirectX/DXIL.td`.
+
+For readabilty, the opcodes can be written in hex instead of the traditional
+decimal value. As an example the 12th experimental op code could be written as
+`0x8000000C` instead of `2147483660`.
```

Comment:
I’m pretty sure we could even do something like `0x80000000 | 12` in TableGen since it has support for arithmetic and bitwise operations with some limitations.